assets/js/base/components/cart-checkout/totals/shipping/calculator-button.tsx

```diff
@@ -5,25 +5,48 @@ import { __ } from '@wordpress/i18n';
 
 export interface CalculatorButtonProps {
 	label?: string;
+	hasRates?: boolean | undefined;
 	isShippingCalculatorOpen: boolean;
 	setIsShippingCalculatorOpen: ( isShippingCalculatorOpen: boolean ) => void;
 }
 
 export const CalculatorButton = ( {
 	label = __( 'Calculate', 'woo-gutenberg-products-block' ),
+	hasRates = true,
 	isShippingCalculatorOpen,
 	setIsShippingCalculatorOpen,
 }: CalculatorButtonProps ): JSX.Element => {
+	// If shipping rates are available. We want to show a button instead of a link.
+	if ( hasRates ) {
+		return (
+			<button
+				className="wc-block-components-totals-shipping__change-address-button"
+				onClick={ () => {
+					setIsShippingCalculatorOpen( ! isShippingCalculatorOpen );
+				} }
+				aria-expanded={ isShippingCalculatorOpen }
+			>
+				{ label }
+			</button>
+		);
+	}
+
+	// If no shipping rates are available, we want to show a link instead of a button.
+	// This is because the style of the link will match the link style of the theme like "Add a coupon".
 	return (
-		<button
-			className="wc-block-components-totals-shipping__change-address-button"
+		<a
+			role="button"
+			href="#wc-block-components-shipping-calculator-address__link"
```

Comment:
This causes the URL in the browser to update when clicking. It is unclear to me why we have to show a link instead of a button when rates are not available, what is the specific reason for showing different elements?

To avoid the URL change you can pass the event as a parameter of the `onClick` handler, and call `.preventDefault()` on it, like so:

```js
onClick={ ( e ) => {
  e.preventDefault();
  setIsShippingCalculatorOpen( ! isShippingCalculatorOpen );
} }
```

Reply:
Ah, I missed adding the ` e.preventDefault();`. Thank you, I have updated the code.  

>It is unclear to me why we have to show a link instead of a button when rates are not available, what is the specific reason for showing different elements?

It's because of the styling of the elements. I kept the original button for the conditions where we are displaying the `calculate` link. 
For others, I'm displaying the `Add an address` link to match the style with the coupon code link. 

I tried using the link at both places but then the `calculate` link gets the theme color which was taking too much attention. So I decided to keep the original style and elements for it. Please let me know if there is a better way to tackle it.